src/main/java/org/manifold/compiler/front/ExpressionEdge.java

```diff
@@ -30,8 +33,7 @@ public void setTarget(ExpressionVertex newTarget) {
 
   public ExpressionEdge(ExpressionVertex from, ExpressionVertex to) {
     if (from == null && to == null) {
-      throw new UndefinedBehaviourError(
-          "attempt to create a totally disconnected ExpressionEdge");
+      log.warn("attempt to create a totally disconnected ExpressionEdge");
     }
     source = from;
     target = to;
```

Comment:
May want to throw an exception if calling setTarget/setSource with null, or if the target/source is already set. Does any code need to call setTarget multiple times?